walk/walk.go

```diff
@@ -210,6 +210,58 @@ func Walk2(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode
 		return errors.Join(w.errs...)
 	}
 
+	// Visit additional directories that extensions requested for indexing.
+	for len(w.relsToVisit) > 0 {
+		// Don't simply range over relsToVisit. We may append more.
+		relToVisit := w.relsToVisit[0]
+		w.relsToVisit = w.relsToVisit[1:]
+
+		dirToVisit := filepath.Join(c.RepoRoot, relToVisit)
+		if fi, err := os.Stat(dirToVisit); err != nil || !fi.IsDir() {
+			// Silently skip non-existant directory.
+			continue
+		}
+
+		// Make sure to visit prefixes of relToVisit as well so we apply
+		// configuration directives.
+		slash := -1
+		for {
+			var rel string
+			i := strings.Index(relToVisit[slash+1:], "/")
+			if i < 0 {
+				rel = relToVisit
+			} else {
+				rel = relToVisit[:slash+1+i]
+				slash = slash + 1 + i
+			}
+
+			if _, ok := w.visits[rel]; !ok {
+				parentRel := path.Dir(rel)
+				if parentRel == "." {
+					parentRel = ""
+				}
+				parentCfg := w.visits[parentRel].c
+				if getWalkConfig(parentCfg).isExcludedDir(rel) {
+					break
+				}
+				c := parentCfg.Clone()
+				w.visit(c, rel, false)
+			}
+
+			if rel == relToVisit {
+				break
+			}
+		}
+
+		if _, ok := w.visits[relToVisit]; ok {
```

Comment:
Can't the for loop above invoke `visit` up to **and including** `relToVisit` instead of requiring this extra `if`+`visit` out here?

Reply:
Yes, removed! Actually it already did, so this was purely redundant.